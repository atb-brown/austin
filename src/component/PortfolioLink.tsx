import Link from "./Link";
import React from "react";

export default function PortfolioLink(props: LinkProps) {
  if (props.issuerDescription && props.issuerUrl) {
    return (
      <LinkWithIssuer
        description={props.description}
        url={props.url}
        urlId={props.urlId}
        issuerDescription={props.issuerDescription}
        issuerUrl={props.issuerUrl}
      />
    );
  }
  return (
    <LinkWithoutIssuer
      description={props.description}
      url={props.url}
      urlId={props.urlId}
    />
  );
}

function LinkWithoutIssuer(props: LinkProps) {
  // TODO: it's a bad idea to have <div/> elements that are children of <p/> elements
  return (
    <div>
      <div>{props.description}</div>
      <div>
        <Link description={props.urlId} url={props.url} />
      </div>
      <br />
    </div>
  );
}

function LinkWithIssuer(props: LinkProps) {
  return (
    <div>
      <div>{props.description}</div>
      <div>
        <Link description={props.urlId} url={props.url} />
      </div>
      <div className="App-soft-text">
        Issued by:{" "}
        <Link
          description={props.issuerDescription ?? ""}
          url={props.issuerUrl ?? ""}
        />
      </div>
      <br />
    </div>
  );
}

type LinkProps = {
  description: string;
  url: string;
  urlId: string;
  issuerDescription?: string;
  issuerUrl?: string;
};
