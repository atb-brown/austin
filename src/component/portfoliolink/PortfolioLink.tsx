import React from "react";
import { PortfolioLinkProps } from "./PortfolioLinkProps";
import LinkWithIssuer from "./PortfolioLinkWithoutIssuer";
import LinkWithoutIssuer from "./PortfolioLinkWithIssuer";

/**
 * A link to some sort of portolio item.
 *
 * @param {PortfolioLinkProps} props Properties.
 * @return {ReactElement} The portfolio link component.
 */
export default function PortfolioLink({
  description,
  url,
  urlId,
  issuerDescription,
  issuerUrl,
}: PortfolioLinkProps) {
  if (issuerDescription && issuerUrl) {
    return (
      <LinkWithIssuer
        description={description}
        issuerDescription={issuerDescription}
        issuerUrl={issuerUrl}
        url={url}
        urlId={urlId}
      />
    );
  }
  return (
    <LinkWithoutIssuer description={description} url={url} urlId={urlId} />
  );
}
