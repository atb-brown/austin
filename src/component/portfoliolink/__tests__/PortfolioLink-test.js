import PortfolioLink from "../PortfolioLink";
import renderer from "react-test-renderer";

it("renders to match snapshot", () => {
  const tree = renderer
    .create(
      <PortfolioLink
        description="description Field"
        url="https://example.com"
        urlId="linkId Field"
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
