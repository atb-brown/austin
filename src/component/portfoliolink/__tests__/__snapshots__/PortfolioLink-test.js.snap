// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match snapshot 1`] = `
<div>
  <div>
    description Field
  </div>
  <div>
    <a
      className="App-link"
      href="https://example.com"
    >
      linkId Field
    </a>
  </div>
  <br />
</div>
`;
